res/controllers/Traktor-Kontrol-S4-MK3.js

```diff
@@ -0,0 +1,3040 @@
+                samplerOrBeatloopRoolPage[i] = new SamplerButton({
+                    number: samplerNumber,
+                });
+                if (samplerCrossfaderAssign) {
+                    engine.setValue(
+                        "[Sampler" + samplerNumber + "]",
+                        "orientation",
+                        (decks[0] === 1) ? 0 : 2
+                    );
+                }
+            }
+            this.keyboard[i] = new KeyboardButton({
+                number: i + 1,
+                deck: this,
+            });
+            i++;
+        }
+
+        const switchPadLayer = (deck, newLayer) => {
+            let index = 0;
+            for (let pad of deck.pads) {
+                pad.outDisconnect();
+                pad.inDisconnect();
+
+                pad = newLayer[index];
+                Object.assign(pad, io.pads[index]);
+                if (!(pad instanceof HotcueButton)) {
+                    pad.color = deck.color;
+                }
+                // don't change the group of SamplerButtons
+                if (!(pad instanceof SamplerButton)) {
+                    pad.group = deck.group;
+                }
+                if (pad.inPacket === undefined) {
+                    pad.inPacket = inPackets[1];
+                }
+                pad.unshift();
+                pad.outPacket = outPacket;
+                pad.inConnect();
+                pad.outConnect();
+                pad.outTrigger();
+                deck.pads[index] = pad;
+                index++;
+            }
+        };
+
+        this.padLayers = {
+            defaultLayer: 0,
+            hotcuePage2: 1,
+            hotcuePage3: 2,
+            samplerPage: 3,
+            keyboard: 5,
+        };
+        switchPadLayer(this, defaultPadLayer);
+        this.currentPadLayer = this.padLayers.defaultLayer;
+
+        this.hotcuePadModeButton = new Button({
+            deck: this,
+            onShortPress: function() {
+                if (!this.shifted) {
+                    if (this.deck.currentPadLayer !== this.deck.padLayers.hotcuePage2) {
+                        switchPadLayer(this.deck, hotcuePage2);
+                        this.deck.currentPadLayer = this.deck.padLayers.hotcuePage2;
+                    } else {
+                        switchPadLayer(this.deck, defaultPadLayer);
+                        this.deck.currentPadLayer = this.deck.padLayers.defaultLayer;
+                    }
+                    this.deck.lightPadMode();
+                } else {
+                    switchPadLayer(this.deck, hotcuePage3);
+                    this.deck.currentPadLayer = this.deck.padLayers.hotcuePage3;
+                    this.deck.lightPadMode();
+                }
+
+            },
+            // hack to switch the LED color when changing decks
+            outTrigger: function() {
+                this.deck.lightPadMode();
+            }
+        });
+        // this.recordPadModeButton = new Button({
```

Comment:
What is this commented out code for? Either delete or add a note why it should be kept.